strawberryfields/tdm/tdmprogram.py

```diff
@@ -605,39 +641,52 @@ def unroll(self, shots):
         # q[sm[2]] as concurrent modes of spatial mode C
         # q[sm[3]] as concurrent modes of spatial mode D.
 
+        # save previous mode index of a command to be able to check when modes
+        # are looped back to the start (not allowed when space-unrolling)
```

Comment:
Is it not allowed or not applicable for space-unrolling?

Reply:
It's not allowed. It'll work to do that here, but it's not physical, and will cause issues with having more applied gates that there should be.